tests/Extractor/HttpClientTest.php

```diff
@@ -290,8 +288,25 @@ public function testTimeout()
         $handler = new TestHandler();
         $logger->pushHandler($handler);
 
-        $guzzle = new GuzzleClient(['defaults' => ['timeout' => 2]]);
-        $adapter = new GuzzleAdapter($guzzle);
+        // find with adapter is installed
+        if (class_exists('Http\Adapter\Guzzle6\Client')) {
+            $guzzle = new \GuzzleHttp\Client(['timeout' => 2]);
+            $adapter = new \Http\Adapter\Guzzle6\Client($guzzle);
+        } elseif (class_exists('Http\Adapter\Guzzle5\Client')) {
+            $guzzle = new \GuzzleHttp\Client(['defaults' => ['timeout' => 2]]);
+            $adapter = new \Http\Adapter\Guzzle5\Client($guzzle);
+        } elseif (class_exists('Http\Client\Curl\Client')) {
+            $adapter = new \Http\Client\Curl\Client(
+                \Http\Discovery\MessageFactoryDiscovery::find(),
+                \Http\Discovery\StreamFactoryDiscovery::find(),
+                [
+                    CURLOPT_TIMEOUT => 2,
+                ]
+            );
+        } else {
+            $this->markTestSkipped('No Guzzle adapter defined ?');
```

Comment:
Else instead of skip, you can do `$adapter = Http\Discovery\HttpClientDiscovery::find();`. So if it need to test with other adapter, it is necessary only to modify travis fail.  Only valid if the 2s of timeout is not a requirements by the tests